### PR TITLE
[DOC-524] CircleCI Dynamic Config

### PR DESCRIPTION
**Pull request by dandimeo**
### Description

The actual CircleCI Config is a static one.
A nice enhancement can be brought by using Dynamic Config feature that CircleCI offers.
Dynamic config will allow to have:
- non-written docs versions, they can be iterated via versions.yml file
- non-written openssl versions to be used to compile feature pr branches
- general flexibility

#### Upstream PRs

- 3.10:
- 3.11: arangodb/enterprise-preview:3.11-nightly
- 3.12: https://github.com/arangodb/arangodb/pull/19666

### CODE REVIEW

File: .circleci/generate_config.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    for i in range(len(versions)):
+        version = versions[i]["name"]
+        
+        compileJob = {
+            "compile-linux": {
+                "context": ["sccache-aws-bucket"],
+                "name": f"compile-{version}",
+                "arangodb-branch": f"arangodb/enterprise-preview:{version}-nightly" if versions[i]["alias"] != "devel" else "arangodb/enterprise-preview:devel-nightly",
+                "version": version,
+                "openssl": "3.0.9",
```
Comment (Simran-B) [inline .circleci/generate_config.py:140]: Is this only a fallback?

Comment (dandimeo) [reply .circleci/generate_config.py:140]: The openssl in this case just serve as a default value for circleci to be able to download the executor docker image (the arangodb-build docker image) to start the compile job.

**Review by nerpaula — APPROVED** [review]

**Pull request merged**